corehq/apps/reports/views.py

```diff
@@ -826,6 +864,21 @@ def send_test_scheduled_report(request, domain, scheduled_report_id):
     return HttpResponseRedirect(reverse("reports_home", args=(domain,)))
 
 
+def _can_send_test_report(report_id, user, domain):
+    try:
+        report = ReportNotification.get(report_id)
+    except ResourceNotFound:
+        return False
+
+    if report.domain != domain:
+        return False
+
+    if user._id != report.owner._id and not user.is_domain_admin(domain):
+        return False
+
+    return True
```

Comment:
Same here.
```suggestion
    return (
        report.domain == domain:
        and user._id == report.owner._id
        and user.is_domain_admin(domain)
    )
```

Alternately could reduce some duplication by reusing the logic above. The simplest thing I can think of given the existing code is:
```suggestion
    return _can_delete_scheduled_report(report, user, domain)
```

Reply:
Yeah, there's definitely a change down the road I'd like to do where this 'must be the owner or a domain admin' is formalized into its own term (perhaps report admin), but I don't think it's worth doing yet -- this logic is also spread all over one of the report templates as well